airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/ReplicationActivityImpl.java

```diff
@@ -217,6 +225,26 @@ private static StandardSyncOutput reduceReplicationOutput(final ReplicationOutpu
     return standardSyncOutput;
   }
 
+  private static void traceReplicationSummary(final ReplicationAttemptSummary replicationSummary) {
+    if (replicationSummary != null) {
+      final Map<String, Object> tags = new HashMap<>();
+      if (replicationSummary.getBytesSynced() != null) {
+        tags.put(REPLICATION_BYTES_SYNCED_KEY, replicationSummary.getBytesSynced());
+        MetricClientFactory.getMetricClient().count(OssMetricsRegistry.REPLICATION_BYTES_SYNCED, replicationSummary.getBytesSynced());
+      }
+      if (replicationSummary.getRecordsSynced() != null) {
+        tags.put(REPLICATION_RECORDS_SYNCED_KEY, replicationSummary.getRecordsSynced());
+        MetricClientFactory.getMetricClient().count(OssMetricsRegistry.REPLICATION_RECORDS_SYNCED, replicationSummary.getRecordsSynced());
+      }
+      if (replicationSummary.getStatus() != null) {
+        tags.put(REPLICATION_STATUS_KEY, replicationSummary.getStatus().value());
+      }
+      if (CollectionUtils.isNotEmpty(tags)) {
```

Comment:
Why not `if (!tags.isEmpty())` instead of bringing in dependency for this one method that simply delegates to calling `!isEmpty` anyways:
```
public static boolean isNotEmpty(@Nullable Collection collection) {
        return collection != null && !collection.isEmpty();
    }
```
As `tags` was created in this block of code, there isn't a chance for it being null.  The null check in the `isNotEmpty` method is not providing anything.

Reply:
> Why not `if (!tags.isEmpty())` instead of bringing in dependency for this one method that simply delegates to calling `!isEmpty` anyways:
> 
> ```
> public static boolean isNotEmpty(@Nullable Collection collection) {
>         return collection != null && !collection.isEmpty();
>     }
> ```
> 
> As `tags` was created in this block of code, there isn't a chance for it being null. The null check in the `isNotEmpty` method is not providing anything.

Will switch.  I added that before I made `tags` not be nullable.  That was the only reason for using it (avoiding a potential NPE).